src/backend/distributed/operations/shard_split.c

```diff
@@ -1145,3 +1221,650 @@ TryDropSplitShardsOnFailure(HTAB *mapOfShardToPlacementCreatedByWorkflow)
 			NULL /* pgResult */);
 	}
 }
+
+
+/*
+ * SplitShard API to split a given shard (or shard group) in non-blocking fashion
+ * based on specified split points to a set of destination nodes.
+ * 'splitOperation'             : Customer operation that triggered split.
+ * 'shardIntervalToSplit'       : Source shard interval to be split.
+ * 'shardSplitPointsList'		: Split Points list for the source 'shardInterval'.
+ * 'workersForPlacementList'	: Placement list corresponding to split children.
+ */
+static void
+NonBlockingShardSplit(SplitOperation splitOperation,
+					  ShardInterval *shardIntervalToSplit,
+					  List *shardSplitPointsList,
+					  List *workersForPlacementList)
+{
+	char *superUser = CitusExtensionOwnerName();
+	char *databaseName = get_database_name(MyDatabaseId);
+	int connectionFlags = FORCE_NEW_CONNECTION;
```

Comment:
Define this close to line 1268 where it is first used.

Reply:
Done.